### DISCUSSION TITLE
What is this trying to accomplish?

### DISCUSSION THREAD

**Discussion by Ughuuu** [TitanNano/godot-rust-script#38] [Q&A] [answered]
I'm confused as to how to use this repo.
I am interested in editing rust files in the editor and having some syntax and possibly also compiling the scripts automatically on save and getting warnings. Is this repo trying to achieve this or not? Also do you have an example/starting point on how to use? I read through the README.md but didn't quite grasp on how to set it up.

Thanks.

**Comment by TitanNano** [comment, marked as answer]
Yes, the README is currently a mess and doesn't match the most recent version of this library anymore. 

Godot Rust provides GDExtensions support with gdext. This basically allows you to create custom node types in a rust library (heavily simplified). 

What this library does is making rust also available as a scripting language via a GDExtension library. There isn't really a significant difference between what you can do with GDExtension classes / custom Nodes and attaching a script to a node, but if you would like to rather attach a script to an existing node, then this library might be able to help you.

The editor support you are writing about is currently not planned, though. Implementing good editor support in Godot would be quite time-consuming, and there are already many editors out there that support rust quite well.

**Reply by TitanNano** [reply]
I added details around this question into the README.
